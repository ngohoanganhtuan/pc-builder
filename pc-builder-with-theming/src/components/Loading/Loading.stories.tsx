// Types
import type { ComponentStory, ComponentMeta } from '@storybook/react';

// Components
import { Loading } from '@components';

export default {
  title: 'PCBuilder/Loading',
  component: Loading,
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} as ComponentMeta<typeof Loading>;

const Template: ComponentStory<typeof Loading> = (args) => <Loading {...args} />;

export const Sample = Template.bind({});

Sample.args = {
  isFetching: true,
};
