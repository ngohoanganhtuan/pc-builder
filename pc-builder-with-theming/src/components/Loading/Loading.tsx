/**
 * @type { Object }
 * @property { boolean } isFetching - Fetching status
 * @property { ReactElement | ReactElement[] } - React children
 */
type Props = {
  isFetching: boolean;
  children: React.ReactNode;
};

/**
 * This component represents Loading
 */
export const Loading = ({ isFetching = true, children }: Props) => {
  if (isFetching) {
    return (
      <div
        className="spinner-border align-self-center m-auto"
        role="status"
      ></div>
    );
  }

  return <>{children}</>;
};
