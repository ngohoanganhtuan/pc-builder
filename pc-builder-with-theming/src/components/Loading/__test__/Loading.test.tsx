import renderer from 'react-test-renderer';
import { Loading } from '../Loading';

describe('Test [Loading] component', () => {
  test('Component [Loading] should render loading if isFetching props is true', () => {
    const component = renderer
      .create(
        <Loading isFetching={true}>
          <h1>Children</h1>
        </Loading>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Component [Loading] should render children if isFetching props is false', () => {
    const component = renderer
      .create(
        <Loading isFetching={false}>
          <h1>Children</h1>
        </Loading>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
