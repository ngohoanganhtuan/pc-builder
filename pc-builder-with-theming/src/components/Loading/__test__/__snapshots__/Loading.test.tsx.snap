// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test [Loading] component Component [Loading] should render children if isFetching props is false 1`] = `
<h1>
  Children
</h1>
`;

exports[`Test [Loading] component Component [Loading] should render loading if isFetching props is true 1`] = `
<div
  className="spinner-border align-self-center m-auto"
  role="status"
/>
`;
